File: 2022/23/java/Main.java
import java.io.IOException;

public class Main {
	private static final String INPUT_FILE = "../input/input.txt";

	public static void main(String[] args) throws IOException {
		Grove grove = new Grove(INPUT_FILE);
		int round = 1;
		while(grove.sim()) {
			if (round++ == 10) {
				int total = grove.emptyCount();
				System.out.format("1. Empty groud tiles: %,d%n", total);
			}
		}
		System.out.format("2. First round with no Elf moves: %,d %n", round);
	}
}

File: 2022/23/rust/src/main.rs
use rust::Grove;

// --- Constants ---
const INPUT_FILE: &'static str = "../input/input.txt";

// --- Main function ---
fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut grove = Grove::load(INPUT_FILE)?;
    let mut round = 1;
    while grove.sim() {
        if round == 10 {
            let total = grove.empty_count();
            println!("1. Empty ground tiles: {}", total);
        }
        round += 1;
    }
    println!("2. First round with no Elf moves: {}", round);
    Ok(())
}
